seatunnel-connectors-v2/connector-http/connector-http-myhours/src/main/java/org/apache/seatunnel/connectors/seatunnel/myhours/source/MyHoursSource.java

```diff
@@ -0,0 +1,102 @@
+import org.apache.seatunnel.connectors.seatunnel.http.source.HttpSource;
+import org.apache.seatunnel.connectors.seatunnel.http.source.HttpSourceReader;
+import org.apache.seatunnel.connectors.seatunnel.myhours.source.config.MyHoursSourceConfig;
+import org.apache.seatunnel.connectors.seatunnel.myhours.source.config.MyHoursSourceParameter;
+
+import org.apache.seatunnel.shade.com.typesafe.config.Config;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.auto.service.AutoService;
+import com.google.common.base.Strings;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Objects;
+
+@Slf4j
+@AutoService(SeaTunnelSource.class)
+public class MyHoursSource extends HttpSource {
+    private final MyHoursSourceParameter myHoursSourceParameter = new MyHoursSourceParameter();
+    @Override
+    public String getPluginName() {
+        return "MyHours";
+    }
+
+    @Override
+    public void prepare(Config pluginConfig) throws PrepareFailException {
+        CheckResult result = CheckConfigUtil.checkAllExists(pluginConfig, MyHoursSourceConfig.EMAIL, MyHoursSourceConfig.PASSWORD);
+        if (!result.isSuccess()) {
+            throw new PrepareFailException(getPluginName(), PluginType.SOURCE, result.getMsg());
+        }
+        //Login to get accessToken
+        String accessToken = null;
+        try {
+            accessToken = getAccessToken(pluginConfig);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        this.myHoursSourceParameter.buildWithConfig(pluginConfig, accessToken);
+        buildSchemaWithConfig(pluginConfig);
+    }
+
+    @Override
+    public AbstractSingleSplitReader<SeaTunnelRow> createReader(SingleSplitReaderContext readerContext) throws Exception {
+        return new HttpSourceReader(this.myHoursSourceParameter, readerContext, this.deserializationSchema);
+    }
+
+    private String getAccessToken(Config pluginConfig) throws IOException {
+        MyHoursSourceParameter myHoursLoginParameter = new MyHoursSourceParameter();
+        myHoursLoginParameter.buildWithLoginConfig(pluginConfig);
+        HttpClientProvider loginHttpClient = new HttpClientProvider(myHoursLoginParameter);
+        try {
+            HttpResponse response = loginHttpClient.doPost(myHoursLoginParameter.getUrl(), myHoursLoginParameter.getBody());
+            if (HttpResponse.STATUS_OK == response.getCode()) {
+                String content = response.getContent();
+                if (!Strings.isNullOrEmpty(content)) {
+                    ObjectMapper om = new ObjectMapper();
+                    Map<String, String> contentMap = om.readValue(content, Map.class);
+                    return contentMap.get(MyHoursSourceConfig.ACCESSTOKEN);
+                }
+            }
+            log.error("login http client execute exception, http response status code:[{}], content:[{}]", response.getCode(), response.getContent());
```

Comment:
If get token failed, I think should throw exception there not print error message, because without token connector will generate data failed.